src/facade/cmd_arg_parser.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "facade/cmd_arg_parser.h"
+
+#include <absl/strings/ascii.h>
+#include <absl/strings/match.h>
+#include <absl/strings/numbers.h>
+
+#include "base/logging.h"
+#include "facade/error.h"
+
+namespace facade {
+
+CmdArgParser::CheckProxy::operator bool() const {
+  if (idx_ >= parser_->args_.size())
+    return false;
+
+  std::string_view arg = parser_->SafeSV(idx_);
+  if ((!ignore_case_ && arg != tag_) || (ignore_case_ && !absl::EqualsIgnoreCase(arg, tag_)))
```

Comment:
Can there ever happen the `tag_` == "" ? If that's the case, we got a small bug here

Reply:
Yes in _theory_, but actually no because command arguments can't be empty